hathor/event/websocket/factory.py

```diff
@@ -0,0 +1,99 @@
+        self.connections_to_stream_events: Dict[str, EventStreaming] = {}
+        super().__init__()
+
+        self.log = logger.new()
+        self.is_running = False
+        self.event_storage = event_storage
+
+        # A timer to periodically send new events to clients that sent start_streaming_events message
+        self._lc_send_events = LoopingCall(self._send_events_to_subscribed_clients)
+        self._lc_send_events.clock = reactor
+
+    def start(self):
+        self.is_running = True
+
+        # Start event sender
+        self._lc_send_events.start(settings.WS_SEND_EVENTS_INTERVAL, now=False)
+
+    def stop(self):
+        self.is_running = False
+
+        if self._lc_send_events.running:
+            self._lc_send_events.stop()
+
+    def on_client_close(self, connection: HathorEventWebsocketProtocol) -> None:
+        """Called when a ws connection is closed."""
+        # A connection closed before finishing handshake will not be in `self.connections`.
+        self.connections.discard(connection)
+        del self.connections_to_stream_events[connection.id]
+
+    def on_client_open(self, connection: HathorEventWebsocketProtocol) -> None:
+        """Called when a ws connection is opened (after handshaking)."""
+        connection.id = str(uuid4())
+        self.connections.add(connection)
+
+    def handle_message(self, connection: HathorEventWebsocketProtocol, data: bytes) -> None:
+        message = json_loadb(data)
+        if message['type'] == 'start_streaming_events':
+            self._handle_start_streaming_events(connection, message)
+        elif message['type'] == 'stop_streaming_events':
+            self._handle_stop_streaming_events(connection, message)
+        elif message['type'] == 'get_event':
+            self._handle_get_event(connection, message)
+
+    def _handle_start_streaming_events(self, connection: HathorEventWebsocketProtocol, message: Dict) -> None:
+        response = {'type': 'start_streaming_events', 'success': True}
+        event_id = message['event_id'] if 'event_id' in message else 0
+
+        if not event_id.isdigit():
+            response['success'] = False
+            response['reason'] = 'event_id must be a positive integer number'
+        else:
+            self.connections_to_stream_events[connection.id] = EventStreaming(connection, event_id)
+
+        payload = json_dumpb(response)
+        connection.sendMessage(payload, False)
+
+    def _handle_stop_streaming_events(self, connection: HathorEventWebsocketProtocol, message: Dict) -> None:
+        self.connections_to_stream_events.pop(connection.id)
+        del self.connections_to_stream_events[connection.id]
+        payload = json_dumpb({'type': 'stop_streaming_events', 'success': True})
+        connection.sendMessage(payload, False)
+
+    def _handle_get_event(self, connection: HathorEventWebsocketProtocol, message: Dict) -> None:
+        pass
+
+    def _send_events_to_subscribed_clients(self) -> None:
```

Comment:
Why we don't send events in real-time?

Reply:
I'll refactor it so it's real time

Reply:
Addressed in [this comment](https://github.com/HathorNetwork/hathor-core/issues/405#issuecomment-1406692278)

Reply:
Done!